v2/yarpcjson/inbound.go

```diff
@@ -42,7 +43,11 @@ type jsonHandler struct {
 func (h jsonHandler) Handle(ctx context.Context, reqBody interface{}) (interface{}, error) {
 	results := h.handler.Call([]reflect.Value{reflect.ValueOf(ctx), reflect.ValueOf(reqBody)})
 	if appErr, _ := results[1].Interface().(error); appErr != nil {
-		return results[0].Interface(), appErr
+		return results[0].Interface(), yarpcerror.New(
+			yarpcerror.CodeUnknown,
+			appErr.Error(),
+			yarpcerror.WithDetails(appErr),
```

Comment:
When the inbound codec has `EncodeError(appError)`, do we expect it to figure out whether `appErr` is serializable as JSON, or should we make that distinction here?

This is cool.

Reply:
@robbertvanginkel

Reply:
The former, I think. The codec should handle all the serialization logic.

Reply:
The `appErr` shouldn't be in the details here, details should be empty if the user didn't supply a yarpcerror.